decode.go

```diff
@@ -70,7 +70,7 @@ func (de *decoder) message(buf []byte, sval reflect.Value) error {
 	// Decode all the fields
 	fields := ProtoFields(sval.Type())
 	fieldi := 0
-	for len(buf) > 0 {
+	for len(buf) > 0 && fieldi < len(fields) {
```

Comment:
> And what exact problem is this proposed check intended to fix?

Let me chime in and explain what from my understanding the original problem was. Under some circumstances, this return statement:
```
rem, err := de.value(wiretype, buf, field)
if err != nil {
     return fmt.Errorf("FieldName %s: %v", fields[fieldi].Name, err)
}
```
(https://github.com/dedis/protobuf/blob/08bf2120a31f8c77ba4307a6a44ee3174f28b973/decode.go#L105)
panic-ed because `fieldi` was out of range and one couldn't see the original error. 
I'm not sure how-to reproduce this behaviour but I guess it's more of an strange encoding (producing invalid protobuf) than a decoding problem (but I didn't think it through).